e2e/getenvoy_run_test.go

```diff
@@ -123,50 +128,69 @@ func envoyRunTest(t *testing.T, test func(*getEnvoy, *adminClient), args ...stri
 
 			// When we get to this line, we can assume Envoy started properly. Run the test
 			if strings.Contains(l, envoyStartedLine) {
-				require.NotEmpty(t, adminAddressPath, "expected adminAddressPath to be set")
-				go runTestAndInterruptEnvoy(t, c, adminAddressPath, test) // don't block stderr when the test is running
+				require.NotEmpty(t, c.runDir, "expected adminAddressPath to be set")
+				requireEnvoyPid(t, c)                        // check here to avoid race condition
+				go runTestAndInterruptEnvoy(ctx, t, c, test) // don't block printing stderr!
 			}
 		}
 	}()
 
 	err = c.cmd.Wait() // This won't hang forever because newGetEnvoy started it with a context timeout!
 	require.NoError(t, err)
 
-	// return the run archive location
-	return c, filepath.Dir(adminAddressPath) + ".tar.gz"
+	// Ensure the Envoy process was terminated
+	_, err = process.NewProcessWithContext(ctx, c.envoyPid) // because os.FindProcess is no-op in Linux!
+	require.Error(t, err, "expected GetEnvoy to terminate Envoy after running [%v]", c)
+
+	return c, func() {
+		// this may not be present if the process was kill -9'd so don't error
```

Comment:
ps we don't yet test kill -9 also kills envoy yet and even if we could probably only linux maybe would support it.